### PR TITLE
MMT-3538: Adds search form to Groups list

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

Adds a search form to the Groups list page

### What areas of the application does this impact?

Searching for Groups

# Testing

Visit the groups page, search for groups by name/provider/members

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/GroupList/GroupList.jsx [hunk 1]
```diff
@@ -114,6 +116,12 @@ const GroupList = () => {
     )
   }, [])
 
+  const buildEllipsisTextCell = useCallback((cellData) => cellData && (
+    <EllipsisText>
+      {cellData}
```
Comment (trevorlang) [inline static/src/js/components/GroupList/GroupList.jsx:121]: Thought about this a little more. I wonder if we should build this into EllipsisText component so we don't have to handle it at this level in any component that uses it.

File: static/src/js/components/Header/Header.jsx [hunk 2]
```diff
@@ -157,6 +158,7 @@ const Header = () => {
                 >
                   User Guide
                 </Button>
+
```
Comment (trevorlang) [inline static/src/js/components/Header/Header.jsx:161]: Not a huge fan of these empty spaces between the components. I think we are a little inconsistent with it but my vote would be no extra spaces 😅 I think thats what we do in most places.

File: static/src/js/pages/GroupListPage/GroupListPage.jsx [hunk 3]
```diff
@@ -62,6 +56,8 @@ const GroupListPage = () => (
     pageType="secondary"
     header={<GroupListPageHeader />}
   >
+    <GroupSearchForm />
+
```
Comment (trevorlang) [inline static/src/js/pages/GroupListPage/GroupListPage.jsx:60]: See extra space rant above ⬆️ 🙃

**Review by trevorlang — APPROVED** [review]
I think it's good to go but have some minor feedback you can take or leave.

**Review by abbottry — APPROVED** [review]

**Pull request merged**